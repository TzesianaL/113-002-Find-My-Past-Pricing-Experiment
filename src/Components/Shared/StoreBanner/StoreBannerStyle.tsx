import styled from 'styled-components'
/**
 * Styled component is used to decorate StoreBanner component.
 * Props are passed from ThemeProvider component.
 */
export const StoreBannerStyle = styled.div`
  background-color: white;
  display: flex;
  width: 100%;
  justify-content: left;
  width: -moz-available;
  align-items: left;
  color: ${(props) => props.theme.color || 'black'} !important;
  font-family: ${(props) => props.theme.fontFamily};
  padding: 0.6rem 0 0.6rem 0;
  padding-left: 1rem;
`
