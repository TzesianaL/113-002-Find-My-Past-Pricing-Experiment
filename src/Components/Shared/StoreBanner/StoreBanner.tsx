import React, { FC } from 'react'
import { StoreBannerStyle } from './StoreBannerStyle'

export interface StoreBannerProps {
  /**
   * Variable name contains the name of the selected store
   */
  name?: string
  /**
   * Variable logoScr contains the src for a logo of the store which will be displayed
   */
  logoSrc?: string
}
/**
 * Component StoreBanner displays the logo of the store on the left corner, if exists, and the store name taking up the whole width of its containing element. Styled components are used to decorate the component.
 */
const StoreBanner: FC<StoreBannerProps> = ({ logoSrc, name }) => {
  if (name === undefined && logoSrc === undefined) {
    throw Error('Store Banner component is used without props (name,logo)')
  }
  return (
    <StoreBannerStyle>
      {logoSrc && <img src={logoSrc} height="44" alt="logo" />}

      {name && <p style={{fontWeight: 'bold', paddingLeft:'0.5rem'}}>{name}</p>}
    </StoreBannerStyle>
  )
}
export default StoreBanner
