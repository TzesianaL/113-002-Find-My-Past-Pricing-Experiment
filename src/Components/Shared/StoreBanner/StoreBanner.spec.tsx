import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import StoreBanner from './StoreBanner'

describe('<StoreBanner/>', () => {
  it('should render correctly Store Banner', () => {
    const tree = renderer
      .create(<StoreBanner name="Storename" logoSrc="logo" />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
