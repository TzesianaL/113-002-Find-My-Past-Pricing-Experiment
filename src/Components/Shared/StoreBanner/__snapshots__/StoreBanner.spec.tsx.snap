// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreBanner/> should render correctly Store Banner 1`] = `
.c0 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fill-available;
  width: -moz-available;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white !important;
  padding: 0.6rem 0 0.6rem 0;
}

.c0 p {
  width: 100%;
  font-size: rem;
}

.c0 img {
  padding: 0 1em;
}

<div
  className="c0"
>
  <img
    alt="logo"
    height="44"
    src="logo"
  />
  <p>
    Storename
  </p>
</div>
`;
